packages/utils/src/signatureUtils.ts

```diff
@@ -1,6 +1,44 @@
 const ethjsutil = require('ethereumjs-util')
 
-export function signMessage(messageHash: string, privateKey: string, address: string) {
+import * as Web3Utils from 'web3-utils'
+
+// If messages is a hex, the length of it should be the number of bytes
+function messageLength(message: string) {
+  if (Web3Utils.isHexStrict(message)) {
+    return (message.length - 2) / 2
+  }
+  return message.length
+}
+// Ethereum has a special signature format that requires a prefix
+// https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_sign
+export function hashMessageWithPrefix(message: string) {
+  const prefix = '\x19Ethereum Signed Message:\n' + messageLength(message)
+  return Web3Utils.soliditySha3(prefix, message)
+}
+
+export function hashMessage(message: string) {
+  return Web3Utils.soliditySha3({ type: 'string', value: message })
+}
+
+// Uses a native function to sign (as signFn), most commonly `web.eth.sign`
+export async function signMessageNatively(
```

Comment:
Maybe `signMessageNoPrefix`?

Reply:
The native signing does prefix the message?

Reply:
I did change `signMessageForMessageHash` to `signMessageNoPrefix`